spec.md

```diff
@@ -1913,6 +1916,78 @@ The CO MUST implement the specified error recovery behavior when it encounters t
 | Operation pending for volume | 10 ABORTED | Indicates that there is a already an operation pending for the specified volume. In general the Cluster Orchestrator (CO) is responsible for ensuring that there is no more than one call "in-flight" per volume at a given time. However, in some circumstances, the CO MAY lose state (for example when the CO crashes and restarts), and MAY issue multiple calls simultaneously for the same volume. The Plugin, SHOULD handle this as gracefully as possible, and MAY return this error code to reject secondary calls. | Caller SHOULD ensure that there are no other calls pending for the specified volume, and then retry with exponential back off. |
 
 
+#### `NodeGetVolumeStats`
+
+A Node plugin MUST implement this RPC call.
+`NodeGetVolumeStats` RPC call returns the volume capacity stats available for the volume.
+
+If the volume is being used in `BlockVolume` mode then `used_capacity` and `available_capacity` MAY be omitted from `NodeGetVolumeStatsResponse`.
+Similarly, inode information may be omitted from `NodeGetVolumeStatsResponse` when unavailable.
+
+
+```protobuf
+message NodeGetVolumeStatsRequest {
+  // The ID of the volume. This field is REQUIRED.
+  string volume_id = 1;
+
+  // The path at which the volume was published. It MUST be an absolute
+  // path in the root filesystem of the process serving this request.
+  // This is a REQUIRED field.
+  string target_path = 2;
+}
+
+message NodeGetVolumeStatsResponse {
+  // The used capacity in bytes. This field is OPTIONAL.
+  // The value of this field MUST NOT be negative.
+  // For volumes that share a filesystem with the host (e.g hostpath)
+  // this is used space on the underlying storage by specified
+  // shared path or directory.
+  int64 used_capacity = 1;
+
+  // The available capacity in bytes. This field is OPTIONAL.
+  // For volumes that share a filesystem with the host (e.g hostpath)
+  // this is available space on the underlying storage and is
+  // shared with the host processes and other volumes.
+  // The value of this field MUST NOT be negative.
+  int64 available_capacity = 2;
+
+  // The total capacity in bytes. This field is REQUIRED.
+  // For volumes that share a file system with the host (e.g hostpath)
+  // this is the total size of underlying storage and it MAY
+  // not be equal to used_capacity + available_capacity because
+  // filesystem is shared.
```

Comment:
I have dropped these elaborate comments, because host paths that are shared with underlying storage may or may not be a strong use case. I expect plugin implementations will use their judgement to return right values from here.